File: DSA/Sarthak_Jain_590014936/DAY 1/Day_1_Question_2.py
#Sarthak Jain 590014936
#Day 1 Question 2
#Find Missing and Repeated Number

lis = []

n = int(input("Enter number of elements:"))
for i in range(n):
    b = int(input("Enter element:"))
    lis.append(b)

print("Given List:")
print(lis)

# Create a count list of size n+1 (index 0 not used)
count = [0] * (n + 1)

for x in lis:
    count[x] += 1

for i in range(1, n + 1):
    if count[i] == 0:
        missing = i
    elif count[i] == 2:
        repeated = i

print("Missing:", missing)
print("Twice:", repeated)

File: DSA/RATNA_KHANNA_590013141/day1question2.c
#include<stdio.h>
int main()
{
    int n;
    printf("enter number of elements in array\n");
    scanf("%d",&n);
    int nums[n];
    printf("enter the elements to the array\n");
    for(int i=0;i<n;i++)
    {
        scanf("%d",&nums[i]);
    }
    int count[n + 1]; 
    for (int i = 0; i <= n; i++) 
    {
        count[i] = 0;
    }
    for (int i = 0; i < n; i++) 
    {
        count[nums[i]]++;
    }
    int missing = -1, repeated = -1;
    for (int i = 1; i <= n; i++) 
    {
        if (count[i] == 0) {
            missing = i;
        } else if (count[i] == 2) {
            repeated = i;
        }
    }
    printf("Missing: %d, Twice: %d\n", missing, repeated);
return 0;
}
